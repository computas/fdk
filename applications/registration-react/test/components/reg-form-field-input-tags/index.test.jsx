import React from 'react';
import { shallow } from 'enzyme';
import InputTagsField from '../../../src/components/reg-form-field-input-tags';

let defaultProps;
let wrapper;

beforeEach(() => {
  defaultProps = {
    showLabel: true,
    input: {
      name: 'input',
      value: 'Datakatalog for RAMSUND OG ROGNAN REVISJON'
    },
    label: 'label',
    type: 'input',
    meta: {
      active: false,
      asyncValidating: false,
      autofilled: false,
      dirty: false,
      form: 'catalog',
      initial: 'Datakatalog for RAMSUND OG ROGNAN REVISJON',
      invalid: false,
      pristine: true,
      submitting: false,
      submitFailed: false,
      touched: false,
      valid: true,
      visited: false
    }
  };
  wrapper = shallow(<InputTagsField {...defaultProps} />);
});

test('should render InputTagsField correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

/*
test('should handle handleChange', () => {
  const mockedEvent = {

  };
  wrapper.find('button').prop('onChange')();
  expect(handleChange).toHaveBeenCalled();
});
*/
