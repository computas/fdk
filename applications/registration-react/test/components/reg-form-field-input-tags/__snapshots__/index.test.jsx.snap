// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render InputTagsField correctly 1`] = `
<div
  className="pl-2"
>
  <label
    className="fdk-form-label w-100"
    htmlFor="input"
  >
    label
    <TagsInput
      addKeys={
        Array [
          9,
          13,
        ]
      }
      addOnBlur={false}
      addOnPaste={false}
      className="fdk-reg-input-tags"
      disabled={false}
      focusedClassName="react-tagsinput--focused"
      inputProps={
        Object {
          "placeholder": "",
        }
      }
      maxTags={-1}
      name="input"
      onChange={[Function]}
      onlyUnique={false}
      pasteSplit={[Function]}
      preventSubmit={true}
      removeKeys={
        Array [
          8,
        ]
      }
      renderInput={[Function]}
      renderLayout={[Function]}
      renderTag={[Function]}
      tagDisplayProp={null}
      tagProps={
        Object {
          "className": "react-tagsinput-tag",
          "classNameRemove": "react-tagsinput-remove",
        }
      }
      validationRegex={/\\.\\*/}
      value="Datakatalog for RAMSUND OG ROGNAN REVISJON"
    />
    <i
      className="fa fa-check-circle fa-lg ml-2 invisible"
    />
  </label>
</div>
`;
